core/src/main/java/fr/sncf/osrd/api/InfraManager.java

```diff
@@ -172,7 +174,8 @@ public FullInfra load(String infraId, String expectedVersion, DiagnosticRecorder
                 // try downloading the infra again if:
                 //  - the existing cache entry hasn't reached a stable state
                 //  - we don't have the right version
-                var obsoleteVersion = expectedVersion != null && !expectedVersion.equals(cacheEntry.version);
+                var obsoleteVersion = expectedVersion != null
+                        && (cacheEntry.version == null || expectedVersion.compareTo(cacheEntry.version) > 0);
```

Comment:
nit: can't you make it more readable with:
```suggestion
                        && (cacheEntry.version == null || expectedVersion > cacheEntry.version);
```